packages/core/src/lib/waku_peer_exchange/index.ts

```diff
@@ -63,13 +65,22 @@ class WakuPeerExchange implements PeerExchange {
         bytes.append(chunk);
       });
 
-      const response = PeerExchangeRPC.decode(bytes).response;
+      const decoded = PeerExchangeRPC.decode(bytes).response;
 
-      if (!response) {
+      if (!decoded) {
         throw new Error("Failed to decode response");
       }
 
-      return response;
+      const peerInfos: PeerInfo[] = [];
+
+      decoded.peerInfos.forEach(async (peerInfo) => {
```

Comment:
`forEach` call will return before any of the internal promises are resolved.

Need to do:

```
await Promise.all(peerInfos.map(async peerInfo => {})
```

To ensure that all promise are resolved.